$NetBSD: patch-aa,v 1.3 2008/02/16 01:24:39 bjs Exp $

If refuse(3) ever supports -ho in option_list, the #ifndef __NetBSD__ 
should be removed.

--- src/wdfs-main.c.orig	2007-04-12 04:30:08.000000000 -0400
+++ src/wdfs-main.c
@@ -131,14 +131,24 @@ static struct fuse_opt wdfs_opts[] = {
 	FUSE_OPT_END
 };
 
+#ifndef FUSE_USE_VERSION
+#define FUSE_USE_VERSION FUSE_VERSION
+#endif
+
 static int wdfs_opt_proc(
 	void *data, const char *option, int key, struct fuse_args *option_list)
 {
 	switch (key) {
 		case KEY_HELP:
 			print_help();
+#ifndef __NetBSD__
+/*
+ * XXX Using refuse(3), this results in abnormal program termination
+ *     with SIG_SEGV.
+ */
 			fuse_opt_add_arg(option_list, "-ho");
 			call_fuse_main(option_list);
+#endif
 			exit(1);
 
 		case KEY_VERSION:
@@ -1213,7 +1223,8 @@ static int wdfs_statfs(const char *local
 
 
 /* just say hello when fuse takes over control. */
-#if FUSE_VERSION >= 26
+
+#if FUSE_USE_VERSION >= 26
 	static void* wdfs_init(struct fuse_conn_info *conn)
 #else
 	static void* wdfs_init()
@@ -1308,7 +1319,7 @@ static void print_help()
 /* just a simple wrapper for fuse_main(), because the interface changed...  */
 static int call_fuse_main(struct fuse_args *args)
 {
-#if FUSE_VERSION >= 26
+#if FUSE_USE_VERSION >= 26
 	return fuse_main(args->argc, args->argv, &wdfs_operations, NULL);
 #else
 	return fuse_main(args->argc, args->argv, &wdfs_operations);
